tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -164,5 +164,38 @@ function test_preserve_scheme() {
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://site.com' ), 'https://site.com' );
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://site.com/blog' ), 'https://site.com/blog' );
 		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://example.org' ), 'https://example.org' );
+
+		// adding www subdomain reverts to original domain
+		$this->assertEquals( Jetpack_Sync_Functions::preserve_scheme( 'banana', 'https://www.example.com', true ), 'https://example.com' );
```

Comment:
I think if the url in the option is set to be` http://`
we should convert `https://www.example.com` to `http://example.com`